### PR TITLE
Adding tests and test dependencies

### PR DESCRIPTION
**Pull request by Poetro**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,16 +3,30 @@
...
+    "karma-mocha": "^0.1.3",
+    "karma-requirejs": "^0.2.5",
+    "karma-sinon-chai": "^0.1.6",
+    "mocha": "^2.4.5",
+    "requirejs": "^2.1.22",
+    "squirejs": "^0.2.1",
+    "underscore": "^1.8.3"
   },
-  "peerDependencies": {
+  "optionalDependencies": {
```
Comment (mstade) [inline package.json]: It's not really optional though, is it? It should probably just be a dependency outright, or replaced altogether with XHR or something.

Comment (Poetro) [inline package.json]: Right, it is not really optional. But it needs quite some work to replace it, but could be done.

Comment (mstade) [inline package.json]: Make the call chief, I'm good with either but if we should keep it as a dependency we shouldn't pretend it's not.

Comment (Poetro) [inline package.json]: @mstade I'm removing the text dependency, and try to solve it with `d3.text` and try to solve the RequireJS build as well to keep the support for `require('d3-utils/append-from-template!./renderer-template.html')`.

Comment (mstade) [inline package.json]: Hmm, seeing as how plugins like these are a very non-standard feature (at least until a proper loader spec exists with such features included) we should probably see if we can remove the plugin usage altogether.

**Comment by FabienDeshayes** [comment]
:+1:

**Comment by gabrielmontagne** [comment]
Superseded by https://github.com/zambezi/d3-utils/pull/2

**Pull request closed without merging**